### ISSUE TITLE
Proposal: add new option `consistent-multiline` to `array-element-newline`

### ISSUE THREAD

**Issue by zypA13510** [eslint/eslint#12066]
**What rule do you want to change?**
[array-element-newline](https://eslint.org/docs/rules/array-element-newline)
**Does this change cause the rule to produce more or fewer warnings?**
more (than consistent) / fewer (than multiline)
see example below
**How will the change be implemented? (New option, new default behavior, etc.)?**
new option
**Please provide some example code that this change will affect:**
```js
let test1 = [
  'a', 'b', {
    c: 'c',
  },
]
let test2 = [
  'a',
  'b',
  'c',
]
```
**What does the rule currently do for this code?**
(:heavy_check_mark: - do not warn :x: - warn)
- `consistent`: test1:heavy_check_mark:test2:heavy_check_mark:
- `{multiline: true}`: test1:x:test2:x:

**What will the rule do after it's changed?**
- `consistent-multiline`: test1:x:test2:heavy_check_mark:

**Are you willing to submit a pull request to implement this change?**
I'm willing to try, no promises though.

**Comment by zypA13510** [comment]
Another approach would be to make `multiline` the third option rather than a direct replacement for `always`, `never` and `consistent`. But it would be a breaking change to existing configurations.
